crux/models/resource.py

```diff
@@ -65,107 +47,114 @@ def __init__(
-        self._media_type = media_type
-        self._modified_at = modified_at
-        self._tags = tags
-        self._folder = None
-        self._labels = labels if labels else {}  # type: Dict[str, str]
-        self._folder = folder
 
+        self._folder = None
+        self.object_model = object_model
         self.connection = connection
-        self.raw_response = raw_response
 
     @property
     def id(self):
         """str: Gets the Resource ID."""
-        return self._id
+        return self.object_model["resourceId"]
 
     @property
     def description(self):
         """str: Gets the Resource Description."""
-        return self._description
+        return self.object_model["description"]
+
+    @description.setter
+    def description(self, description):
+        self.object_model["description"] = description
 
     @property
     def media_type(self):
         """str: Gets the Resource Description."""
-        return self._media_type
+        return self.object_model["mediaType"]
 
     @property
     def dataset_id(self):
         """str: Gets the Dataset ID."""
-        return self._dataset_id
+        return self.object_model["datasetId"]
 
     @property
     def folder_id(self):
         """str: Gets the Folder ID."""
-        return self._folder_id
+        return self.object_model["folderId"]
 
     @property
     def storage_id(self):
         """str: Gets the Storage ID."""
-        return self._storage_id
+        return self.object_model["storageId"]
 
     @property
     def name(self):
         """str: Gets the Resource Name."""
-        return self._name
+        return self.object_model["name"]
+
+    @name.setter
+    def name(self, name):
+        self.object_model["name"] = name
 
     @property
     def config(self):
         """str: Gets the config."""
-        return self._config
+        return self.object_model["config"]
+
+    @config.setter
+    def config(self, config):
+        self.object_model["config"] = config
 
     @property
     def provenance(self):
         """str: Gets the Provenance."""
-        return self._provenance
+        return self.object_model["provenance"]
+
+    @provenance.setter
+    def provenance(self, provenance):
+        self.object_model["provenance"] = provenance
 
     @property
     def type(self):
         """str: Gets the Resource Type."""
-        return self._type
+        return self.object_model["type"]
 
     @property
     def tags(self):
         """:obj:`list` of :obj:`str`: Gets the Resource Tags."""
-        return self._tags
+        return self.object_model["tags"]
+
+    @tags.setter
+    def tags(self, tags):
+        self.object_model["tags"] = tags
 
     @property
     def labels(self):
         """dict: Gets the Resource labels."""
-        return self._labels
+        labels = {}
+        for label in self.object_model["labels"]:
+            labels[label["labelKey"]] = label["labelValue"]
+        return labels
```

Comment:
Could there be a setter for `labels` that does the reverse? So they could do `file.labels = {"k1": "v1", "k2": "v2"}`?